src/main/java/com/ericsson/ei/handlers/MongoDBHandler.java

```diff
@@ -279,56 +324,60 @@ public void createTTLIndex(String dataBaseName, String collectionName, String fi
 		}
 	}
 
-    private MongoCollection<Document> getMongoCollection(String dataBaseName, String collectionName) {
+    private MongoCollection<Document> getMongoCollection(final String dataBaseName, final String collectionName) {
         if (mongoClient == null)
-            return null;
+            return null;  
+
+        MongoCollection<Document> collection = null;
         MongoDatabase db;
-        List<String> collectionList;
         try {
             db = mongoClient.getDatabase(dataBaseName);
-            collectionList = db.listCollectionNames().into(new ArrayList<String>());
+            collection = db.getCollection(collectionName);
+        }
+        catch (final IllegalArgumentException e) {
+            LOGGER.debug("Could not get collection as the name was {} Illegal: {}", collectionName, e);
+            closeMongoDbConecction();
```

Comment:
I think it was closeMongoDbConnection function that caused mongoQueury object to become null when there is connections problem to MongoDb and EI stops query MongoDB even thou MongoDbClient connection is restored. In EI 3.x this function was removed and it solves this Mongo query issue after connections goes down and come back.
And getCollection() failures should not close/remove the MongoDb objects.